code_snippets/java/src/main/java/usecases/microservices/Idempotency.java

```diff
@@ -0,0 +1,25 @@
+package usecases.microservices;
+
+import static dev.restate.sdk.client.CallRequestOptions.*;
+
+import dev.restate.sdk.client.Client;
+
+class Config {
+  public static String RESTATE_URL = "http://localhost:8080";
+}
+
+public class Idempotency {
+
+  // <start_here>
+  Client rs = Client.connect(Config.RESTATE_URL);
+
+  public void reserveProduct(String productId, String reservationId) {
+    // <mark_1>
+    ProductServiceClient.fromClient(rs, productId)
+        .send()
+        .reserve(DEFAULT.withIdempotency(reservationId));
```

Comment:
Use qualified name, it helps understanding this without reading the import:

```suggestion
        .reserve(CallRequestOptions.DEFAULT.withIdempotency(reservationId));
```